### ISSUE TITLE
Lightning: Separate the `mydumper.csv.header` Config Parameter Functionalities

### ISSUE THREAD

**Issue by dsdashun** [pingcap/tidb#40839]
## Enhancement
Currently, the parameter `mydumper.csv.header` has two functionalities tightly coupled:  
1. it treats the first row of the CSV file as header, and maps the header fields names as the target table column names
2. it skips the first row when importing

These two functionalities are tightly coupled:  when set it to `true`, column names of the first row are auto mapped, and the first row is automatically skipped; when set it to `false`, column names of the first row are not mapped, and the first row is also considered as data.  
In some situations, the users just want to skip the first line of the CSV data, but not to map the first line's field names to table column names.  
For example, some CSV data are very big, and the header fields cannot match target table's column names.  Users may be reluctant to change the header's field names.  In this situation, they can just set `mydumper.csv.header` to false, and also set the config to skip the first row.  Then the data can be imported using Lightning without problems.  

In summary, setting `header` to `true` implies skipping the first row.  However, setting `header` to `false` doesn't necessarily mean keeping the first row as data.  This should be controlled by the users.

**Comment by dsdashun** [comment]
After further consideration, the goal is actually to provide a mechanism to skip the first N rows in a data file. Skipping the first row is just a specific scenario for this requirement.
Initially, I intended to add a parameter, such as `skip-rows`, to enable Lightning to skip specific number of rows. However, this requires further consideration for the interface design:
* If a `skip-rows` is given a single integer value, this means that all source files will skip the same number of rows, which is not ideal.
* To address this issue, the parameter should accept a mapping of 'file_name' to 'skip count' so that different files can skip different numbers of rows. However, this configuration can be difficult to use with many different data files.
* An alternative is to provide a pattern as the key to filter files, but this also presents a challenge as users would need to learn how to write the pattern.

The following is a plan to address this requirement for the first step: for CSV configurations, a `skip-header-row` parameter will be provided. If set to `true`, all CSV files in the table will skip the first parsed row. If `header` in CSV configuration is set to `true`, it implies that `skip-header-row` is also true.  Then the meaning of `header` in CSV config is just to specify whether to use the first row as a 'header row' to parse the specific column names for each column.   
To implement this feature, it will still be made possible to skip N rows for specific data files. When `skip-header-row` is set to true, N will be set to 1.
Once the configuration interface has a clear design, it will be easy to parse the configuration and determine the correct N for a data file and to pass to the relevant logic.

**Comment by lance6716** [comment]
> If a skip-rows is given a single integer value, this means that all source files will skip the same number of rows, which is not ideal.

I can only imagine one type of not ideal situation, which is we plan to import multiple tables but only one table need to be skipped first row. But for that situation, we users can split tables into two groups based in skip N rows. Is the single parameter acceptable for this situation or there may exists other situation?

**Comment by dsdashun** [comment]
> or there may exists other situation

Consider the scenario where a user has split a large data source into 100 files for a single table import. Previously, the first 1000 lines of `file001` and the first 500 lines of `files002` to `file100` were imported into `tableA`. Now, the user wants to reuse the same data source, skipping the imported rows, to import into a new table, `tableB`.

**Comment by dsdashun** [comment]
After further consideration, I plan to introduce a new config parameter `skip-header` in the CSV configuration. The difference between the `header` config is as follows:

* If `header` is set to `true`, it indicates that the first line of the CSV file is a header line. By default, the field names of the header line are mapped to the corresponding column names in the table.
* The `skip-header` config only takes effect when `header` is set to `true`. If `skip-header` is set to `true`, it means that the header line in the CSV file is ignored and not parsed as the column name for the table.

In this context, the `skip-header` config parameter only affects the handling of the header line in CSV files, not the actual data rows. The issue of skipping a specific number of data rows is separate and can be addressed in a separate issue.

**Comment by dsdashun** [comment]
The final version, the parameter name is changed , because `header` and `skip-header` is a bit hard to understand when they are compared.  Now we use `header-schema-match` to indicate that the column names of the CSV header can match those in the target table definitions.  By default, it is set to 'true'.  If it is set to 'false', it indicates that the columns in the CSV header might not have exact matches in the target table. In this scenario, Lightning will ignore the CSV header to avoid errors.
